File: binarySearch.java
import java.util.*;
class binarySearch{
    int  binary(int a[],int n,int key){
        int l,h,mid;
        l=0;
        h=n;
        while(l<=h){
            mid = (l+h)/2;
            if(a[mid]<key){
                l=mid+1;
            }
            else if(a[mid]==key){
                return mid;
            }
            else{
                h=mid-1;
            }
        }
        return -1;

    }
    public static void main(String args[]){
        int n,i,key;
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter Size of Array: ");
        n=sc.nextInt();
        System.out.print("Enter Elements of Array: ");
        int a[]=new int[n];
        for(i=0;i<n;i++){
            a[i]=sc.nextInt();
        }
        System.out.print("Enter Key: ");
        key = sc.nextInt();
        binarySearch ob=new binarySearch();
        int ans = ob.binary(a, n, key);
        System.out.print("Your key is present at "+ans+" Index.");

    }
}

File: Binary_Search.py
def Binary(n,l,key):
    s=0
    e=n
    while s<=e:
        mid = int((s+e)/2)
        if l[mid]==key:
            return f"Your Key is present at {mid} Index"
        elif l[mid]>key:
            s=mid+1
        else:
            e=mid-1
            
    return "Your Key is not present in List"         

def main():
    try:
        n=int(input("Enter the size of List: "))
        l = []
        print("Enter Elements of List: ")
        for i in range(0,n):
            a=int(input())
            l.append(a)
        print(l)
        key = int(int(input("Enter Key: ")))
        result = int(Binary(n,l,key))
        print(result)
    except Exception as e:
        print(e)
if __name__=='__main__':
    main()